libcaf_core/src/detail/meta_object.cpp

```diff
@@ -54,7 +54,26 @@ span<const meta_object> global_meta_objects() {
 const meta_object* global_meta_object(type_id_t id) {
   if (id < meta_objects_size) {
     auto& meta = meta_objects[id];
-    return !meta.type_name.empty() ? &meta : nullptr;
+    if (!meta.type_name.empty())
+      return &meta;
+  }
+  CAF_CRITICAL_FMT(
+    "found no meta object for type ID %d!\n"
+    "        This usually means that run-time type initialization is missing.\n"
+    "        With CAF_MAIN, make sure to pass all custom type ID blocks.\n"
+    "        With a custom main, call (before any other CAF function):\n"
+    "        - caf::core::init_global_meta_objects()\n"
+    "        - <module>::init_global_meta_objects() for all loaded modules\n"
+    "        - caf::init_global_meta_objects<T>() for all custom ID blocks",
+    static_cast<int>(id));
+  return nullptr;
```

Comment:
Is this unreachable code? Which to me would mean a compiler warning and it should not be here.

Reply:
The function could also return a reference if it's marked `[[noreturn]]` given that `CAF_CRITICAL` always ends up calling `std::abort()`.

Reply:
ACK to both.